packages/react-components/source/react/library/drawer/Drawer.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { bool, string, element, node, oneOf } from 'prop-types';
+import Button from '../button';
+
+const propTypes = {
+  /** Control whether the body content is open or closed with the open boolean prop. */
+  open: bool,
+  /** Button text used when the drawer is toggled open. */
+  buttonTextOpen: string,
+  /** Button text used when the drawer is toggled closed. */
+  buttonTextClosed: string,
+  /** Content that is displayed within the drawer header and is always visible */
+  headerContent: element,
+  /** Lets you pick between transparent and text. */
+  buttonType: oneOf(['transparent', 'text']),
+  children: node,
+};
+
+const defaultProps = {
+  open: undefined,
+  buttonTextOpen: 'Hide details',
+  buttonTextClosed: 'Details',
+  headerContent: undefined,
+  buttonType: 'transparent',
+  children: undefined,
+};
+
+function Drawer({
+  open,
+  buttonTextOpen,
+  buttonTextClosed,
+  headerContent,
+  buttonType,
+  children,
+}) {
+  const [isOpen, setIsOpen] = useState(open);
+
+  const shouldSetIsOpen = () =>
+    open === undefined ? setIsOpen(!isOpen) : setIsOpen(open);
```

Comment:
i left in the open prop so that the drawer can be controlled. But if its not being controlled the defaultOpen prop will work and a user can specify if they want the first render to be with the drawer open or not.